src/state/experiments.js

```diff
@@ -61,3 +155,50 @@ export const load = () => dispatch => {
       dispatch({ type: LOAD_ERROR, error: e });
     });
 };
+
+export const showEditModal = (show, experimentID = null) => dispatch => {
+  const type = show ? OPEN_EDIT_MODAL : CLOSE_EDIT_MODAL;
+  return dispatch({ type, experimentID });
+};
+
+export const create = (name, description) => dispatch => {
+  dispatch({ type: CREATE });
+  return api
+    .createExperiment(name, description)
+    .then(res => {
+      dispatch({ type: CREATE_SUCCESS });
+      dispatch(load());
+      dispatch(showEditModal(true, res.id));
```

Comment:
here is a good example why it's better to keep store view agnostic. `create` action can be called from anywhere. But it will set the modal show to true. It is super confusing and can cause weird bugs.

Reply:
Changed in https://github.com/src-d/code-annotation/pull/195/commits/ce357ffaf5ea919e448cbcd238e52032ed6373cd